File: 프로그래머스/0/181934. 조건 문자열/조건 문자열.py
# 간단한 식
def solution(ineq, eq, n, m):
    return int(eval(str(n) + ineq + eq.replace("!", "") + str(m)))


# 내 풀이
# def solution(ineq, eq, n, m):
#     answer = 0
    
#     if ineq == "<":
#         if eq == "=":
#             if n <= m:
#                 answer += 1
#             else:
#                 answer += 0
#         elif eq == "!":
#             if n < m:
#                 answer += 1
#             else:
#                 answer += 0
            
#     elif ineq == ">":
#         if eq == "=": 
#             if n >= m:
#                 answer += 1
#             else:
#                 answer += 0
#         elif eq == "!":
#             if n > m:
#                 answer += 1
#             else:
#                 answer += 0
                
#     return answer

File: 프로그래머스/0/181934. 조건 문자열/조건 문자열.java
class Solution {
    public int solution(String ineq, String eq, int n, int m) {
        if(ineq.equals("<")){
            if(eq.equals("=")){
                // 조건식 ? 참일 때 반환할 값 : 거짓일 때 반환할 값;
                return n <= m ? 1 : 0; 
            }
            else{
                return n < m ? 1 : 0;
            }
        }
        else if(ineq.equals(">")){
            if(eq.equals("=")){
                return n >= m ? 1 : 0;
            }
            else{
                return n > m ? 1 : 0;
            }
        }
        // 기본 반환값을 추가하여 예상치도 못한 ineq 값에 대해 처리
        return 0;
    }
}